core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -990,11 +990,19 @@ private[spark] object Utils extends Logging {
     for ((key, value) <- extraEnvironment) {
       environment.put(key, value)
     }
+
+    // When testing, expose the parent class path to the child. This is processed by
+    // compute-classpath.{cmd,sh} and makes all needed jars available to child processes
+    // when the assembly is built with the "*-provided" profiles enabled.
+    if (sys.props.contains("spark.testing")) {
+      environment.put("SPARK_DIST_CLASSPATH", sys.props("java.class.path"))
```

Comment:
But `SPARK_DIST_CLASSPATH` is not set in the parent (the parent being the surefire/scalatest runner, in this case). The goal of this code is to add the classpath built by those (which includes all the dependencies from the pom) to the child processes without having to modify the test code to manually do it.